### ISSUE TITLE
f1_with_ner

### ISSUE THREAD

**Issue by WangSheng21s** [thunlp/PL-Marker#36]
首先感谢作者出色的工作,有个小小的疑问问您：

为什么在运行关系抽取任务中，在验证集中f1_with_ner的结果能够达到1.0呀，
难道运用的是对应的golden ner嘛，如果是的话能否指出对应代码在run_re.py中的位置，我看好像用的是模型预测的结果做的呀，但是按道理应该不可能到1.0.

谢谢

**Comment by YeDeming** [comment]
请问你运行的指令是?

**Comment by WangSheng21s** [comment]
GPU_ID=0  # For ALBERT-xxlarge, change learning_rate from 2e-5 to 1e-5     # SciERC,  --use_ner_results: use the original entity type predicted by NER models mkdir scire_models for seed in 42; do CUDA_VISIBLE_DEVICES=$GPU_ID  python3  run_re.py  --model_type bertsub  \
    --model_name_or_path  ../bert_models/scibert_scivocab_uncased  --do_lower_case  \
    --data_dir scierc  \
    --learning_rate 2e-5  --num_train_epochs 10 --per_gpu_train_batch_size 8 --per_gpu_eval_batch_size 16 --gradient_accumulation_steps 1 \
    --max_seq_length 256 --max_pair_length 16 --save_steps 1000 \
    --do_train  --do_eval  --evaluate_during_training   --eval_all_checkpoints  --eval_logsoftmax  \
    --fp16  --seed $seed      \
    --test_file sciner_models/PL-Marker-scierc-scibert-44/ent_pred_test.json  \
    --use_ner_results \
    --output_dir scire_models/scire-scibert-$seed  --overwrite_output_dir done; # Average the scores python3 sumup.py scire scire-scibert
​运行命令如上

​
![image](https://user-images.githubusercontent.com/102408340/195521377-d8a8ab70-8a90-4227-93c9-65a3276abad2.png)


运行结果如图，谢谢

**Comment by WangSheng21s** [comment]
上面命令有点乱，补充一个图
![image](https://user-images.githubusercontent.com/102408340/195521558-cfb0a325-106a-42f7-a8e1-1f0fb7dab881.png)

**Comment by YeDeming** [comment]
因为这里测试的用的--dev_file是golden的，如果需要测试end2end的话需要先用ner模型预测一份ner结果
···
mkdir sciner_models
for seed in 42 43 44 45 46; do 
CUDA_VISIBLE_DEVICES=$GPU_ID  python3  run_acener.py  --model_type bertspanmarker  \
    --model_name_or_path  bert_models/scibert-uncased  --do_lower_case  \
    --data_dir scierc  \
    --learning_rate 2e-5  --num_train_epochs 50  --per_gpu_train_batch_size  8  --per_gpu_eval_batch_size 16  --gradient_accumulation_steps 1  \
    --max_seq_length 512  --save_steps 2000  --max_pair_length 256  --max_mention_ori_length 8    \
    --do_train  --do_eval  --evaluate_during_training   --eval_all_checkpoints  \
    --fp16  --seed $seed  --onedropout  --lminit  \
    --train_file train.json --dev_file dev.json --test_file test.json  \
    --output_dir sciner_models/PL-Marker-scierc-scibert-$seed  --overwrite_output_dir  --output_results
done;
···
得到一份带ner预测结果的 ent_pred_dev.json 作为--dev_file的参数

**Comment by YeDeming** [comment]
稍微改了下进入条件，这里是输出ner结果的代码，由于我们是先ner再re，re依赖于ner模型
https://github.com/thunlp/PL-Marker/blob/07fde08d868134ced1d861d17d263d6c782bb420/run_acener.py#L765-L781

**Comment by WangSheng21s** [comment]
好的谢谢
